extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/index/ChooseHeuristicMatchIndexQueryStrategy.java

```diff
@@ -56,12 +54,15 @@ public boolean hasNext() {
 						else {
 							double totalMax = 0;
 							for (final MultiDimensionalNumericData qr : queryRanges) {
-								totalMax += getCostLogic(
+								final double[] dataRangePerDimension = new double[qr.getDimensionCount()];
+								for (int d = 0; d < dataRangePerDimension.length; d++) {
+									dataRangePerDimension[d] = qr.getMaxValuesPerDimension()[d] - qr.getMinValuesPerDimension()[d];
+								}
+								totalMax += IndexUtils.getDimensionalBitsUsed(
 										nextIdx.getIndexStrategy(),
-										nextIdx.getIndexModel(),
-										qr);
+										dataRangePerDimension);
 							}
-							if (totalMax < min) {
+							if (totalMax > min) {
```

Comment:
See prior comment.  This max makes sense to me IF we are trying to maximize the unused bits.
These reads: i want to maximize bits used.  But the most constraining dimension is going to least amount of bits.  Perhaps I am missing something.